### PR TITLE
support read_kwargs_provider arg to compact_partition

### PR DESCRIPTION
**Pull request by raghumdani**
Motivation #124

### CODE REVIEW

File: deltacat/compute/compactor/utils/io.py [hunk 1]
```diff
@@ -282,4 +284,4 @@ def limit_input_deltas(
     logger.info(f"Hash bucket count: {hash_bucket_count}")
     logger.info(f"Input uniform delta count: {len(rebatched_da_list)}")
 
-    return rebatched_da_list, hash_bucket_count, high_watermark
+    return rebatched_da_list, hash_bucket_count, high_watermark, require_multiple_rounds
```
Comment (rkenmi) [inline deltacat/compute/compactor/utils/io.py:287]: Minor: Could we also update the function return type signature?

File: deltacat/compute/compactor/compaction_session.py [hunk 2]
```diff
@@ -163,7 +168,9 @@ def _execute_compaction_round(
     rebase_source_partition_high_watermark: Optional[int],
     enable_profiler: Optional[bool],
     metrics_config: Optional[MetricsConfig],
-    list_deltas_kwargs=Optional[Dict[str, Any]],
+    list_deltas_kwargs: Optional[Dict[str, Any]],
+    read_kwargs_provider: Optional[ReadKwargsProvider],
```
Comment (rkenmi) [inline deltacat/compute/compactor/compaction_session.py:172]: Thanks for adding. Have you thought about allowing clients to pass in various ReadKwargsProviders for specific steps (hash bucketing, materialize)?

**Review by rkenmi — APPROVED** [review]

**Review by valiantljk — APPROVED** [review]

**Comment by valiantljk** [comment]
LGTM, thanks for the update

File: deltacat/compute/compactor/compaction_session.py (hunk 2 above)
Comment (raghumdani) [reply deltacat/compute/compactor/compaction_session.py:172]: Yeah, thought about that. Assuming, we have to read the pk the same way (enforcing schema) we read during materialize, I kept it as one arg for now. If we find a need we can add if required in future.

**Pull request merged**